tls/s2n_handshake.h

```diff
@@ -204,6 +204,7 @@ struct s2n_handshake {
 
 extern message_type_t s2n_conn_get_current_message_type(struct s2n_connection *conn);
 extern int s2n_conn_set_handshake_type(struct s2n_connection *conn);
+S2N_RESULT s2n_conn_set_tls13_handshake_type(struct s2n_connection *conn);
```

Comment:
We could make this function static and call it within `s2n_conn_set_handshake_type`  when the connection is TLS13 version, I don't see a necessity to make this globally accessible.